src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class Duke {
+    public static ArrayList<Task> tasks = new ArrayList<Task>();
+    public static final Scanner SCANNER = new Scanner(System.in);
+
+    public static void main(String[] args) {
+        String userInput = "";
+        Output.printStart();
+
+        Storage.readFile(tasks);
+
+        while (true) {
+            userInput = SCANNER.nextLine();
+            Output.printBorder();
+            if (userInput.trim().equals("bye")) {
+                Storage.writeToFile(tasks);
+                Output.printBye();
+                break;
+            } else if (userInput.startsWith("list")) {
+                System.out.println("Here are the tasks in your list: ");
+                for (int i = 0; i < tasks.size(); i++) {
+                    System.out.print(i + 1 + ".");
+                    System.out.println(tasks.get(i));
+                }
+            } else if (userInput.split(" ")[0].equals("done")) {
+                if(userInput.length() < 6) {
```

Comment:
Here, the literal 6 might not make sense to the reader in the first instant, perhaps try and refactor this magic literal into a constant. This can similarly be done in other instances of magic literals. 

https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers